src/prototype_rust_sdk/workflow_future.rs

```diff
@@ -214,22 +233,30 @@ impl Future for WorkflowFuture {
                                     ),
                                 );
                             }
-                            CancellableID::ChildWorkflow(seq) => {
-                                let start_request = self
-                                    .child_workflow_starts
-                                    .get(&seq)
-                                    .expect("Tried to cancel unknown child");
+                            CancellableID::ChildWorkflow {
+                                cancel_cmd_seq,
+                                child_seq,
```

Comment:
I don't really understand the logic here, we want to cancel the child, why don't we use `child_seq`?

Reply:
I will address all this with the cancel external wf machine, which will end up looking a lot like this one.